Update redirects in login sapling

RyanLassigBanks opened the pull request:
Update how the login sapling redirects a user to the page they were
attempting to navigate to. This is to make it compatible with updates
being made to Canopyjs which will now redirect users to the login page
if they're not logged in. Canopyjs redirects will cause the history to
contain the page the user requested so login simply needs to return that
user to that page after they have been logged in.

Signed-off-by: Ryan Banks <rbanks@bitwise.io>

RyanLassigBanks committed:
Update redirects in login sapling

Update how the login sapling redirects a user to the page they were
attempting to navigate to. This is to make it compatible with updates
being made to Canopyjs which will now redirect users to the login page
if they're not logged in. Canopyjs redirects will populate a field in
`$CANOPY` called `redirected_from`, so login simply needs to return that
user to that page after they have been logged in.

Signed-off-by: Ryan Banks <rbanks@bitwise.io>

dplumb94 reviewed (changes requested):
This appears to have fixed the bug where you can access a sapling before you are logged in which is great. I am seeing some strange functionality where I navigate to a page while not logged in (say /profile). I am redirected to the login sapling as expected and my path is updated to /login. However, once I do log in, I'm brought to the main page, but my path says /profile even though the profile sapling is not rendered. I tested this PR in tandem with the CanopyJS PR so the bug could be in either component.

Also, I wonder if there is any way to avoid the Canopy sidebar rendering at all before the user is logged in, instead of appearing to redirect/reload to the login page.

RyanLassigBanks committed:
Login no longer responsible for calling hideCanopy

Hiding the Canopy navbar is now handled by canopyjs before
redirecting unauthorized users to the login page.

Signed-off-by: Ryan Banks <rbanks@bitwise.io>

RyanLassigBanks commented:
>Also, I wonder if there is any way to avoid the Canopy sidebar rendering at all before the user is logged in, instead of appearing to redirect/reload to the login page.

Added commit removing call to hideCanopy from login sapling and updated canopy js commit to call hideCanopy before redirecting to login page

RyanLassigBanks commented:
>This appears to have fixed the bug where you can access a sapling before you are logged in which is great. I am seeing some strange functionality where I navigate to a page while not logged in (say /profile). I am redirected to the login sapling as expected and my path is updated to /login. However, once I do log in, I'm brought to the main page, but my path says /profile even though the profile sapling is not rendered. I tested this PR in tandem with the CanopyJS PR so the bug could be in either component.

I changed how the redirects were being handled to no longer use `history.goBack` as was the case in my previous commit, so I think I've managed to fix this.

eloaverona reviewed (commented):
Tested and seems to work.  Just one comment.

eloaverona commented on saplings/register-login/src/index.ts:
```diff
@@ -34,6 +34,16 @@ interface FormEventHandler {
   (this: HTMLFormElement, event: Event): void;
 }
 
+interface Canopy {
```
Is this really necessary here? I think it makes more sense as part of the splinter ui app itself, where $CANOPY is initially defined in `splinter-ui/app/src/App.js`

eloaverona reviewed (approved):
Tested
